data/postRouter.js

```diff
@@ -64,10 +64,38 @@ router.put('/:id', (req, res) => {
             res.status(200).json(updated)
         }
     }).catch(error => {
-        res.status(500).json({ message: 'The post information could not be modified.' })
+        res.status(500).json({ error: 'The post information could not be modified.' })
+    })
+})
+
+router.get('/:id/comments', (req, res) => {
+    const id = req.params.id;
+    
+    Posts.findPostComments(id).then(comments => {
+        if(comments && comments.length){
+            res.status(200).json(comments)
+        } else {
+            res.status(404).json({ message: "The post with the specified ID does not exist." })
+        }
+    }).catch(error => {
+        res.status(500).json({ error: 'The comments information could not be retrieved.' })
     })
 })
 
+router.post('/:id/comments', (req, res) => {
+    let comment = req.body;
+    const id = req.params.id;
+    comment.post_id = id;
 
+    if (!comment.text){
+        res.status(404).json({ message: "Please provide text in the comment." })
```

Comment:
I think this should be a 400 error. You'll want to check for post by id here too, otherwise you could end up trying to add comments for a post that doesn't exist which would most likely throw a 500 error